owncloudApp/src/main/java/com/owncloud/android/ui/preview/PreviewVideoActivity.kt

```diff
@@ -0,0 +1,438 @@
+import com.owncloud.android.R
+import com.owncloud.android.databinding.VideoPreviewBinding
+import com.owncloud.android.domain.files.model.OCFile
+import com.owncloud.android.domain.utils.Event
+import com.owncloud.android.extensions.filterMenuOptions
+import com.owncloud.android.extensions.sendDownloadedFilesByShareSheet
+import com.owncloud.android.presentation.authentication.AccountUtils
+import com.owncloud.android.presentation.files.operations.FileOperation.SetFilesAsAvailableOffline
+import com.owncloud.android.presentation.files.operations.FileOperation.UnsetFilesAsAvailableOffline
+import com.owncloud.android.presentation.files.operations.FileOperationsViewModel
+import com.owncloud.android.presentation.files.removefile.RemoveFilesDialogFragment.Companion.newInstance
+import com.owncloud.android.presentation.previews.PreviewVideoViewModel
+import com.owncloud.android.presentation.transfers.TransfersViewModel
+import com.owncloud.android.ui.activity.FileActivity
+import com.owncloud.android.ui.activity.FileDisplayActivity
+import com.owncloud.android.ui.dialog.ConfirmationDialogFragment
+import com.owncloud.android.ui.fragment.FileFragment
+import com.owncloud.android.ui.preview.PrepareVideoPlayerAsyncTask.OnPrepareVideoPlayerTaskListener
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.launch
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import timber.log.Timber
+
+@OptIn(UnstableApi::class)
+class PreviewVideoActivity : FileActivity(), Player.Listener, OnPrepareVideoPlayerTaskListener, FileFragment.ContainerActivity, MenuProvider {
+    private var mAccount: Account? = null
+
+    private lateinit var playerView: PlayerView
+    private var player: ExoPlayer? = null
+    private lateinit var trackSelector: DefaultTrackSelector
+    private var mAutoplay = true
+    private var mPlaybackPosition: Long = 0
+    private var windowInsetsController: WindowInsetsControllerCompat? = null
+    private val previewVideoViewModel: PreviewVideoViewModel by viewModel()
+    private val fileOperationsViewModel: FileOperationsViewModel by viewModel()
+    private val transfersViewModel: TransfersViewModel by viewModel()
+
+    private lateinit var binding: VideoPreviewBinding
+    override fun onCreate(savedInstanceState: Bundle?) {
+        requestWindowFeature(Window.FEATURE_ACTION_BAR_OVERLAY)
+        super.onCreate(savedInstanceState)
+        binding = VideoPreviewBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        addMenuProvider(this)
+
+        // set layoutInDisplayCutoutMode to NEVER so that it never renders content in the cutout area (with devices having notches)
+        // necessary if going in and out immersive mode
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            window.attributes.layoutInDisplayCutoutMode = WindowManager.LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_NEVER;
+        }
```

Comment:
Yes, not in manfiest but in the style that this Activity is using. Also I did that and since this change require API level 27 it is asking me to add that value to styles -v27, let me know If I should do this in that file otherwise for now I did in the `styles.xml` and it is fine (but with the said warning).